test/ng/directive/inputSpec.js

```diff
@@ -1196,7 +1196,7 @@ describe('input', function() {
 
       it('should validate if max is empty', function() {
         $rootScope.maxVal = undefined;
-        $rootScope.value = new Date(9999, 11, 31, 23, 59, 59);
+        $rootScope.value = new Date(3000, 11, 31, 23, 59, 59);
```

Comment:
I didn't feel like an explanation was required in the actual code, so I put it on the commit message instead. The problem is, I don't really know why MS Edge doesn't like the original Data, nor could I reproduce it independently (outside of the tests), but I figured it is not worth investigating further.